common/optimiser/Optimiser.java

```diff
@@ -78,8 +83,12 @@ public boolean isError() {
         return status == Status.ERROR;
     }
 
+    private boolean constraintsChanged() {
+        return constraints.stream().anyMatch(c -> !c.isInitialised);
+    }
+
     private void initialiseSolver() {
-        solver = MPSolver.createSolver("SCIP");
+        solver = MPSolver.createSolver("SAT");
```

Comment:
worth a comment on why/what this change is!!

Also worth mentionining in the PR description's second section

Reply:
You're right!
We switch from a mixed integer programming solver to an integer linear programming solver that is optimised around the variables and coefficients being integers.